### PR TITLE
Exasol: `CREATE/DROP/ALTER USER/ROLE` clean up for consistency

### PR DESCRIPTION
**Pull request by tunetheweb**
### Brief summary of the change made

Noticed while reviewing #3043 that these statements were inconsistent with other dialects.

### Are there any other side effects of this change that we should be aware of?
Shouldn't be

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_exasol.py [hunk 1]
```diff
@@ -2332,13 +2332,13 @@ class FileOptionSegment(BaseSegment):
 ############################
 # USER
 ############################
-class CreateUserSegment(BaseSegment):
+class CreateUserStatementSegment(BaseSegment):
```
Comment (barrywhart) [inline src/sqlfluff/dialects/dialect_exasol.py:2335]: Is it worth using inheritance for some of these segment types? Then you'd be inheriting the `type` property, as well as all the `is_...` properties.

**Review by barrywhart — APPROVED** [review]

File: src/sqlfluff/dialects/dialect_exasol.py (hunk 1 above)
Comment (tunetheweb) [reply src/sqlfluff/dialects/dialect_exasol.py:2335]: `CreateUser` doesn't exist in ANSI (maybe it should) but `CreateRole` does.

BTW this was one advantage if the `replace=True` decorator - was more obviously which ones were overriding and which weren't.

Comment (tunetheweb) [reply src/sqlfluff/dialects/dialect_exasol.py]: We have a `Drop User` in ANSI so I added a `CREATE USER` too. Which allows this to be inherited now.

Comment (barrywhart) [reply src/sqlfluff/dialects/dialect_exasol.py]: Re: `replace`, what if we provided a `@new_segment` decorator that you **have** to use on new segments, and anything else is assumed to be (and automatically verified to be) a replacement?

Comment (tunetheweb) [reply src/sqlfluff/dialects/dialect_exasol.py]: Not sure it's worth the effort. It's not a big deal. Was just a little handy to know that.

**Pull request merged**